osu.Game/Rulesets/Mods/DifficultyAdjustSettingsControl.cs

```diff
@@ -16,38 +17,30 @@ public class DifficultyAdjustSettingsControl : SettingsItem<float?>
         [Resolved]
         private IBindable<WorkingBeatmap> beatmap { get; set; }
 
-        /// <summary>
-        /// Used to track the display value on the setting slider.
-        /// </summary>
-        /// <remarks>
-        /// When the mod is overriding a default, this will match the value of <see cref="Current"/>.
-        /// When there is no override (ie. <see cref="Current"/> is null), this value will match the beatmap provided default via <see cref="updateCurrentFromSlider"/>.
-        /// </remarks>
-        private readonly BindableNumber<float> sliderDisplayCurrent = new BindableNumber<float>();
-
-        protected override Drawable CreateControl() => new SliderControl(sliderDisplayCurrent);
-
         /// <summary>
         /// Guards against beatmap values displayed on slider bars being transferred to user override.
         /// </summary>
         private bool isInternalChange;
 
-        private DifficultyBindable difficultyBindable;
+        private readonly DifficultyBindableWithCurrent current = new DifficultyBindableWithCurrent();
 
         public override Bindable<float?> Current
         {
-            get => base.Current;
-            set
-            {
-                // Intercept and extract the internal number bindable from DifficultyBindable.
-                // This will provide bounds and precision specifications for the slider bar.
-                difficultyBindable = (DifficultyBindable)value.GetBoundCopy();
-                sliderDisplayCurrent.BindTo(difficultyBindable.CurrentNumber);
-
-                base.Current = difficultyBindable;
-            }
+            get => current.Current;
+            set => current.Current = value;
         }
 
+        /// <summary>
+        /// Used to track the display value on the setting slider.
+        /// </summary>
+        /// <remarks>
+        /// When the mod is overriding a default, this will match the value of <see cref="Current"/>.
+        /// When there is no override (ie. <see cref="Current"/> is null), this value will match the beatmap provided default via <see cref="updateCurrentFromSlider"/>.
+        /// </remarks>
+        private BindableNumber<float> sliderDisplayCurrent => current.DisplayBindable;
```

Comment:
What purpose does this serve? Can the usages of this use `current.DisplayBindable` directly, rather than going through one more layer of indirection (accompanied by one more confusing name to grasp, to boot)?

Reply:
Just to make it clear in usages that `current.DisplayBindable` is the slider bindable, with the name `DisplayBindable`, I think it won't be that confusing to inline it in usages, but not sure if I were to revert it to `CurrentNumber`